From f68419e746455d3c7eb6d95a4a1da7a6f7a5c505 Mon Sep 17 00:00:00 2001
From: RawScape <6241315+RawScape@users.noreply.github.com>
Date: Tue, 16 Jun 2026 19:08:25 +0200
Subject: [PATCH] [TMDb] Fix duplicate include_adult parameter in search
 (#1995)

The include_adult parameter was appended to the search query
twice, causing TMDb to reject requests with HTTP 400.

Fixes #1992
---
 src/scrapers/movie/tmdb/TmdbMovieSearchJob.cpp | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/src/scrapers/movie/tmdb/TmdbMovieSearchJob.cpp b/src/scrapers/movie/tmdb/TmdbMovieSearchJob.cpp
index 226becd872..fa6d2c08f7 100644
--- a/src/scrapers/movie/tmdb/TmdbMovieSearchJob.cpp
+++ b/src/scrapers/movie/tmdb/TmdbMovieSearchJob.cpp
@@ -46,10 +46,7 @@ void TmdbMovieSearchJob::doStart()
         url.swap(newUrl);
 
     } else {
-        QUrl newUrl(m_api.getMovieSearchUrl(searchStr,
-            config().locale,
-            config().includeAdult,
-            TmdbApi::UrlParameterMap{{TmdbApi::ApiUrlParameter::INCLUDE_ADULT, includeAdult}}));
+        QUrl newUrl(m_api.getMovieSearchUrl(searchStr, config().locale, config().includeAdult, {}));
         url.swap(newUrl);
         QVector<QRegularExpression> rxYears;
         rxYears << QRegularExpression(R"(^(.*) \((\d{4})\)$)") << QRegularExpression("^(.*) (\\d{4})$")
@@ -65,8 +62,7 @@ void TmdbMovieSearchJob::doStart()
                 QUrl newSearchUrl = m_api.getMovieSearchUrl(searchTitle,
                     config().locale,
                     config().includeAdult,
-                    TmdbApi::UrlParameterMap{{TmdbApi::ApiUrlParameter::INCLUDE_ADULT, includeAdult},
-                        {TmdbApi::ApiUrlParameter::YEAR, searchYear}});
+                    TmdbApi::UrlParameterMap{{TmdbApi::ApiUrlParameter::YEAR, searchYear}});
                 url.swap(newSearchUrl);
                 break;
             }
